### PR TITLE
Cleanup MemAccessUtils.

### PR DESCRIPTION
**Pull request by atrick**
Organize the utilities in this file by section to make subsequent
diffs easier to read and prepare for adding more utilities.

Purely code shuffling to make it possible to stage in other changes.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test

File: include/swift/SIL/MemAccessUtils.h [hunk 1]
```diff
@@ -406,107 +418,53 @@ class AccessedStorage {
   bool operator==(const AccessedStorage &) const = delete;
   bool operator!=(const AccessedStorage &) const = delete;
 };
-} // end namespace swift
+} // namespace swift
 
 namespace llvm {
-
+using swift::AccessedStorage;
```
Comment (gottesmm) [inline include/swift/SIL/MemAccessUtils.h:424]: Question. Does this inject swift::AccessedStorage into llvm for anyone who includes this header?

File: include/swift/SIL/MemAccessUtils.h [hunk 2]
```diff
@@ -406,107 +418,53 @@ class AccessedStorage {
   bool operator==(const AccessedStorage &) const = delete;
   bool operator!=(const AccessedStorage &) const = delete;
 };
-} // end namespace swift
+} // namespace swift
```
Comment (gottesmm) [inline include/swift/SIL/MemAccessUtils.h:421]: Can you put a space here in between 420 and 421?

File: include/swift/SIL/MemAccessUtils.h [hunk 3]
```diff
@@ -765,6 +792,20 @@ Result AccessUseDefChainVisitor<Impl, Result>::visit(SILValue sourceAddr) {
     }
   };
 }
+} // end namespace swift
```
Comment (gottesmm) [inline include/swift/SIL/MemAccessUtils.h:795]: Same here.

File: lib/SIL/Utils/MemAccessUtils.cpp [hunk 4]
```diff
@@ -419,6 +345,125 @@ bool swift::memInstMustInitialize(Operand *memOper) {
   }
 }
 
+bool swift::isSingleInitAllocStack(AllocStackInst *asi,
+                                   SmallVectorImpl<Operand *> &destroyingUses) {
+  // For now, we just look through projections and rely on memInstMustInitialize
```
Comment (gottesmm) [inline lib/SIL/Utils/MemAccessUtils.cpp:350]: Are all the changes in this file, just moving things around? Without substantial changes?

**Review by gottesmm — COMMENTED** [review]
Quick questions.

File: lib/SIL/Utils/MemAccessUtils.cpp (hunk 4 above)
Comment (atrick) [reply lib/SIL/Utils/MemAccessUtils.cpp:350]: Purely code shuffling to make it possible to stage in other changes.

**Comment by atrick** [comment]
Recleaned with all @gottesmm 's comments

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
Thanks!

**Pull request merged**